src/app/shared/header/header.component.ts

```diff
@@ -0,0 +1,38 @@
+import { Component } from '@angular/core';
+import { Router } from '@angular/router';
+import { AuthService } from '../../services/auth.service';
+import { SessionService } from '../../services/session.service';
+
+@Component({
+  selector: 'app-header',
+  templateUrl: './header.component.html',
+  styleUrls: ['./header.component.scss']
+})
+export class HeaderComponent {
+  user: Object;
+  error: string = '';
+
+  constructor(
+    private router: Router,
+    private auth: AuthService,
+    private session: SessionService
+  ) {
+    this.user = this.session.getSession();
+  }
+
+  isLoggedIn() {
+    return this.session.isLoggedIn;
```

Comment:
Shouldn't `return this.session.isLoggedIn` be invoked here? All you are doing is returning a function declaration here and not the result of calling that function.